src/typeahead/highlight.ts

```diff
@@ -1,11 +1,15 @@
 import {Component, Input, OnChanges, ChangeDetectionStrategy, SimpleChanges} from '@angular/core';
 import {regExpEscape, toString} from '../util/util';
 
+/**
+ * Directive that can be used inside a custom result template in order to highlight the term inside the text of the
```

Comment:
Technically it's a component, so it makes it look a bit confusing:

![screen shot 2018-08-03 at 10 15 04](https://user-images.githubusercontent.com/8074436/43632349-24ff37cc-9706-11e8-8d1e-5c31e0482466.png)

Reply:
Changed to Component and merged.

Reply:
FYI, @pkozlowski-opensource and @maxokorokov, I just reused the documentation conventions from other components. I looked at progressbar, carousel, datepicker, rating: they're all components, and their documentation describes them as a directive.

Reply:
Good to know, I'll do a pass through all documentation comments when fixing #2498